internal/rpc/starknet/starknet.go

```diff
@@ -65,161 +69,138 @@ func (s *StarkNetRpc) GetStateUpdate(blockId *BlockId) (any, error) {
+		return nil, UnexpectedError
 	}
 	return NewTxn(tx)
 }
 
-func (s *StarkNetRpc) GetTransactionByBlockIdAndIndex(blockId *BlockId, index uint64) (any, error) {
-	b, err := getBlockById(blockId, s.blockManager)
+func (s *StarkNetRpc) GetTransactionByBlockIdAndIndex(blockId *BlockId, index *uint64) (any, error) {
+	b, err := getBlockById(blockId, s.blockManager, s.logger)
 	if err != nil {
 		return nil, err
 	}
-	if index >= b.TxCount {
-		return nil, NewInvalidTxnIndex()
+	if index == nil || *index >= b.TxCount {
+		return nil, InvalidTxnIndex
 	}
-	txHash := b.TxHashes[index]
+	txHash := b.TxHashes[*index]
 	tx, err := s.txnManager.GetTransaction(txHash)
 	if err != nil {
-		// TODO: manage unexpected error
+		s.logger.Errorw(err.Error(), "function", "GetTransactionByBlockIdAndIndex")
+		return nil, UnexpectedError
 	}
 	return NewTxn(tx)
 }
 
-func (s *StarkNetRpc) GetTransactionReceipt(transactionHash string) (any, error) {
-	// Parsing TxnHash param
-	if !isFelt(transactionHash) {
-		return nil, NewInvalidTxnHash()
-	}
-	txHash := new(felt.Felt).SetHex(transactionHash)
-	_receipt, err := s.txnManager.GetReceipt(txHash)
+func (s *StarkNetRpc) GetTransactionReceipt(transactionHash *RpcFelt) (any, error) {
+	receipt, err := s.txnManager.GetReceipt(transactionHash.Felt())
 	if err != nil {
 		if errors.Is(err, db.ErrNotFound) {
-			return nil, NewInvalidTxnHash()
+			return nil, InvalidTxnHash
 		}
-		// TODO: manage unexpected error
+		s.logger.Errorw(err.Error(), "function", "GetTransactionReceipt")
+		return nil, UnexpectedError
 	}
-	return NewReceipt(_receipt)
+	return NewReceipt(receipt)
 }
 
-func (s *StarkNetRpc) GetClass(classHash string) (any, error) {
-	// Parsing ClassHash param
-	if !isFelt(classHash) {
-		return nil, NewInvalidContractClassHash()
-	}
-	_ = new(felt.Felt).SetHex(classHash)
+func (s *StarkNetRpc) GetClass(classHash *RpcFelt) (any, error) {
 	_, latestBlockHash := s.synchronizer.LatestBlockSynced()
 	latestBlock, err := s.blockManager.GetBlockByHash(latestBlockHash)
 	if err != nil {
-		// TODO: manage unexpeceted error
-		return nil, errors.New("unexpected error")
+		if errors.Is(err, db.ErrNotFound) {
+			return nil, InvalidContractClassHash
+		}
+		s.logger.Errorw(err.Error(), "function", "GetClass")
+		return nil, UnexpectedError
 	}
 	_ = state.New(s.stateManager, latestBlock.NewRoot)
 	// TODO: implement class service
-	return nil, errors.New("unimplemented")
+	return nil, NotImplementedError
 }
 
-func (s *StarkNetRpc) GetClassHashAt(blockId *BlockId, address string) (any, error) {
-	if !isFelt(address) {
-		return nil, NewContractNotFound()
-	}
-	addressF := new(felt.Felt).SetHex(address)
-	b, err := getBlockById(blockId, s.blockManager)
+func (s *StarkNetRpc) GetClassHashAt(blockId *BlockId, address *RpcFelt) (any, error) {
+	b, err := getBlockById(blockId, s.blockManager, s.logger)
 	if err != nil {
-		return nil, NewInvalidBlockId()
+		return nil, err
```

Comment:
Shouldn't this return an `InvalidBlockId` error?

Reply:
The `getBlockById` function returns the proper error